modules/core/test/image/testImageWarpAffine.cpp

```diff
@@ -0,0 +1,288 @@
+/****************************************************************************
+ *
+ * This file is part of the ViSP software.
+ * Copyright (C) 2005 - 2017 by Inria. All rights reserved.
+ *
+ * This software is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * ("GPL") version 2 as published by the Free Software Foundation.
+ * See the file LICENSE.txt at the root directory of this source
+ * distribution for additional information about the GNU GPL.
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See http://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Description:
+ * Test for vpImageTools::warpAffine() function.
+ *
+ * Author:
+ * Vikas Thamizharasan
+ *
+ *****************************************************************************/
+/*!
+  \example testImageWarpAffine.cpp
+
+  \brief Test the warpAffine function with different types of transformation matrix
+*/
+
+#include <visp3/gui/vpDisplayD3D.h>
+#include <visp3/gui/vpDisplayGDI.h>
+#include <visp3/gui/vpDisplayGTK.h>
+#include <visp3/gui/vpDisplayOpenCV.h>
+#include <visp3/gui/vpDisplayX.h>
+#include <visp3/io/vpImageIo.h>
+#include <visp3/io/vpParseArgv.h>
+
+#include <visp3/core/vpIoTools.h>
+#include <visp3/core/vpImageTools.h>
+
+// List of allowed command line options
+#define GETOPTARGS "cdip:h"
+
+
+namespace {
+  /*
+    Print the program options.
+
+    \param name : Program name.
+    \param badparam : Bad parameter name.
+    \param ipath: Input image path.
+    \param ppath: Personal image path.
+   */
+  void usage(const char *name, const char *badparam, std::string ipath, std::string ppath)
+  {
+    fprintf(stdout, "\n\
+  Test image warping.\n\
+  \n\
+  SYNOPSIS\n\
+    %s [-i <input image path>] [-p <personal image path>] [-c] [-d]\n\
+       [-h]\n                 \
+  ", name);
+
```

Comment:
Should these be the input parameters or could the user instead input the `angle`, `scale`(x,y) and `translation`(x,y) ? Should it throw an error if one of the parameter is missing or assume default value as 0?